src/lib/y2firewall/clients/proposal.rb

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2firewall/firewalld/api"
+require "y2firewall/proposal_settings"
+require "y2firewall/dialogs/proposal"
+require "installation/proposal_client"
+
+module Y2Firewall
+  module Clients
+    class Proposal < ::Installation::ProposalClient
+      include Yast::I18n
+      include Yast::Logger
+
+      attr_accessor :settings, :firewalld
+
+      SERVICES_LINKS = [
+        LINK_ENABLE_FIREWALL = "firewall--enable_firewall".freeze,
+        LINK_DISABLE_FIREWALL = "firewall--disable_firewall".freeze,
+        LINK_OPEN_SSH_PORT = "firewall--open_ssh".freeze,
+        LINK_BLOCK_SSH_PORT = "firewall--close_ssh".freeze,
+        LINK_ENABLE_SSHD = "firewall--enable_sshd".freeze,
+        LINK_DISABLE_SSHD = "firewall--disable_sshd".freeze,
+        LINK_OPEN_VNC = "firewall--open_vnc".freeze,
+        LINK_CLOSE_VNC = "firewall--close_vnc".freeze,
+      ]
+
+      LINK_FIREWALL_DIALOG = "firewall".freeze
+
+      def initialize
+        Yast.import "UI"
+        Yast.import "HTML"
+        textdomain "installation"
+
+        @firewalld = Y2Firewall::Firewalld.new
+        @settings ||= ProposalSettings.instance
+      end
+
+    protected
+
+      def description
+        {
+          # Proposal title
+          "rich_text_title" => _("Firewall and SSH"),
+          # Menu entry label
+          "menu_title"      => _("&Firewall and SSH"),
+          "id"              => LINK_FIREWALL_DIALOG
+        }
+      end
+
+      def make_proposal(attrs)
```

Comment:
I do not see the `force_reset`. Is it still needed?

Reply:
It was not handled in the original client, I could at it reloading the defaults.